homeassistant/components/group/entity.py

```diff
@@ -425,6 +425,19 @@ def _async_update_group_state(self, tr_state: State | None = None) -> None:
         elif tr_state.attributes.get(ATTR_ASSUMED_STATE):
             self._assumed_state = True
 
+        registry: GroupIntegrationRegistry = self.hass.data[REG_KEY]
+
+        def _active_on_states() -> set[str]:
+            return {
+                state.state
+                for entity_id in self.tracking
+                if (domain := entity_id.split(".")[0]) in registry.on_states_by_domain
+                and (state := self.hass.states.get(entity_id)) is not None
+                and state.state in registry.on_states_by_domain[domain]
+            }
+
+        active_domains = {entity_id.split(".")[0] for entity_id in self.tracking}
```

Comment:
Maybe generate this when we change `tracking` so we don't have to do it every time?

Reply:
Might be able to set something like the following mutually exclusive

`self.single_active_domain = `  
`self.active_domains = `

Than you can probably avoid some checks if `single_active_domain` is set